test/distributed/checkpoint/test_file_system_checkpoint_cpu.py

```diff
@@ -433,6 +433,7 @@ def test_switch_between_sharded_tensor_to_tensor(self, thread_count) -> None:
                     "sharded": sharded_tensor.rand(save_spec, tensor_size),
                     "replicated": torch.rand(tensor_size, device="cpu"),
                 }
+                dist.broadcast(save_dict["replicated"], src=0)
```

Comment:
Curious: did you see different result from different ranks without this line?

Reply:
Yep! iirc correctly, in the original code the `replicated` key actually had a different value on each rank. The test passed previously because the `_dedup_tensors` method always kept the value in the first rank. In the new code, the value is kept on whichever rank is storing the least amount of data, so it could have been any of the ranks.